### PR TITLE
[ISSUE-909] Clean spam logs of node removal and maintenance in every reconcile

### PR DESCRIPTION
**Pull request by CraneShiEMC**
Signed-off-by: Shi, Crane <crane.shi@emc.com>

### Issue dell/csi-baremetal#909

Clean spam logs of nonexistent node removal and maintenance in every reconcile

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

### CODE REVIEW

File: pkg/nodeoperations/controller.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func (c *Controller) handleNodeRemoval(ctx context.Context, csibmnodes []nodecrd
 		errors        []string
 		removingNodes []nodecrd.Node
 	)
-	c.log.Info("Starting Node Removal")
+	c.log.Debug("Starting Node Removal")
```
Comment (rkiselenko) [inline pkg/nodeoperations/controller.go:117]: This log entry should be moved in actual Node Removal routine.

**Review by rkiselenko — CHANGES_REQUESTED** [review]

File: pkg/nodeoperations/controller.go (hunk 1 above)
Comment (CraneShiEMC) [reply pkg/nodeoperations/controller.go]: > This log entry should be moved in actual Node Removal routine.

Has changed the code according to your comment.

Comment (rkiselenko) [reply pkg/nodeoperations/controller.go]: You dont get, this log entry should be moved in the part of code there is actual removal process happen. I think it should be in removeNodes method:

```golang
func (c *Controller) removeNodes(ctx context.Context, csibmnodes []nodecrd.Node) error {
  if len(csibmnodes) > 0 {
    c.log.Debugf("Starting Node Removal, node count: %d",  len(csibmnodes))
  }
// ....
```
As you can see log entry shows only then there are any nodes to remove exist. The same changes should be this `Starting Node Maintenance`. 

After this changes both log entries will not spam log.

Comment (CraneShiEMC) [reply pkg/nodeoperations/controller.go]: > You dont get it, this log entry should be moved in the part of code there is actual removal process happen. I think it should be in removeNodes method:
> 
> ```go
> func (c *Controller) removeNodes(ctx context.Context, csibmnodes []nodecrd.Node) error {
>   if len(csibmnodes) > 0 {
>     c.log.Debugf("Starting Node Removal, node count: %d",  len(csibmnodes))
>   }
> // ....
> ```
> 
> As you can see log entry shows only then there are any nodes to remove exist. The same changes should be with `Starting Node Maintenance` log.
> 
> After this changes both log entries will not spam log.

OK. I will change the code according to your comment.

**Review by rkiselenko — APPROVED** [review]

**Review by yurkov-anton — APPROVED** [review]

**Pull request merged**